pkgs/test_core/lib/src/bootstrap/vm.dart

```diff
@@ -24,3 +27,30 @@ void internalBootstrapVmTest(Function Function() getMain, SendPort sendPort) {
     platformChannel.sink.add('done');
   });
 }
+
+/// Bootstraps a native executable test to communicate with the test runner over
+/// a socket.
+void internalBootstrapNativeTest(
+    Function Function() getMain, List<String> args) async {
+  if (args.length != 2) {
+    throw StateError(
+        'Expected exactly two args, a host and a port, but got $args');
+  }
+  var socket = await Socket.connect(args[0], int.parse(args[1]));
+  var platformChannel = MultiChannel<Object?>(StreamChannel(socket, socket)
+      .cast<List<int>>()
```

Comment:
There is probably a better way of doing this, so please feel free to suggest something :), same thing exists in the platform code.